src/libcore/xml.cpp

```diff
@@ -1174,6 +1174,17 @@ ref<Object> load_string(const std::string &string, const std::string &variant,
 
 ref<Object> load_file(const fs::path &filename_, const std::string &variant,
                       ParameterList param, bool write_update) {
+
+    ref<Thread> thread = Thread::thread();
+    ref<FileResolver> fr = thread->file_resolver();
+    ref<FileResolver> fr2 = new FileResolver(*fr);
+    thread->set_file_resolver(fr2);
+
+    // Add the scene file's directory to the search path.
+    fs::path scene_dir = filename_.parent_path();
+    if (!fr2->contains(scene_dir))
+        fr2->append(scene_dir);
+
```

Comment:
Could you explain a bit why this is needed?

Reply:
This is not really related to the `dict_to_xml` implementation, but more of a fix of an issue I encountered when creating unit tests. A given scene XML file may include relative paths to textures, meshes... Therefore we need to add the scene folder to the file resolver, otherwise it won't be able to find these files if we try to load the scene from outside this folder. This was previously done in mitsuba's main function before calling `load_file`, thus ensuring that calling `mitsuba path/to/scene.xml` would work independently of the current working directory. However, this meant that directly calling `load_file` from python would fail, which is why I moved this directly in `load_file`.

Reply:
I think previously those lines where written in the python script explicitly. See the [doc](https://mitsuba2.readthedocs.io/en/latest/src/python_interface/parsing_xml.html#loading-a-scene):
```
# Absolute or relative path to the XML file
filename = 'path/to/my/scene.xml'

# Add the scene directory to the FileResolver's search path
Thread.thread().file_resolver().append(os.path.dirname(filename))

# Load the scene for an XML file
scene = load_file(filename)
```

Reply:
Why not do it directly in `load_file` then?